internal/provider/workflow_job_resource.go

```diff
@@ -0,0 +1,320 @@
+		return
+	}
+
+	client, ok := req.ProviderData.(*AAPClient)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *AAPClient, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Schema defines the schema for the  workflowjobresource.
+func (r *WorkflowJobResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"inventory_id": schema.Int64Attribute{
+				Optional: true,
+				Computed: true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+				Description: "Identifier for the inventory the job will be run against. " +
+					"If not provided, the job will run against the default inventory.",
+			},
+			"workflow_job_template_id": schema.Int64Attribute{
+				Required:    true,
+				Description: "Id of the workflow job template.",
+			},
+			"job_type": schema.StringAttribute{
+				Computed:    true,
+				Description: "Job type",
+			},
+			"url": schema.StringAttribute{
+				Computed:    true,
+				Description: "URL of the workflow job template",
+			},
+			"status": schema.StringAttribute{
+				Computed:    true,
+				Description: "Status of the workflow job",
+			},
+			"extra_vars": schema.StringAttribute{
+				Description: "Extra Variables. Must be provided as either a JSON or YAML string.",
+				Optional:    true,
+				CustomType:  customtypes.AAPCustomStringType{},
+			},
+			"triggers": schema.MapAttribute{
+				Optional:    true,
+				ElementType: types.StringType,
+				Description: "Map of arbitrary keys and values that, when changed, will trigger a creation" +
+					" of a new Workflow Job on AAP. Use 'terraform taint' if you want to force the creation of" +
+					" a new workflow job without changing this value.",
+			},
+			"ignored_fields": schema.ListAttribute{
+				ElementType: types.StringType,
+				Computed:    true,
+				Description: "The list of properties set by the user but ignored on server side.",
+			},
+		},
+		MarkdownDescription: "Launches an AAP workflow job.\n\n" +
+			"A workflow job is launched only when the resource is first created or when the " +
+			"resource is changed. The " + "`triggers`" + " argument can be used to " +
+			"launch a new workflow job based on any arbitrary value.\n\n" +
+			"This resource always creates a new workflow job in AAP. A destroy will not " +
+			"delete a workflow job created by this resource, it will only remove the resource " +
+			"from the state.",
+	}
+}
+
+func (r *WorkflowJobResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data WorkflowJobResourceModel
+
+	// Read Terraform plan data into workflow job resource model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
```

Comment:
I realize this function signature is implementing an interface, so there is no way to return an error with contextual information. But with only checking that an error exists then returning are we losing helpful information about where an error occurred, or is the expectatation that the `resp.Diagnostics` contains the error context and code higher up will emit a log message?